packages/flutter_tools/lib/src/commands/run.dart

```diff
@@ -135,6 +135,11 @@ abstract class RunCommandBase extends FlutterCommand with DeviceBasedDevelopment
               'this comma separated list of allowed prefixes.',
         valueHelp: 'skia.gpu,skia.shaders',
       )
+      ..addFlag('enable-dart-profiling',
+        defaultsTo: true,
+        help: 'Whether the Dart VM sampling CPU profiler is enabled. This flag '
```

Comment:
is this different from the cpu profiler that can be toggled in devtools?

Reply:
It is very likely the same CPU profiler, although I've never seen/discovered the toggle to turn it off in DevTools. I suppose instead of adding this flag, I could start up DevTools to turn off the CPU profiler, and then start up the Observatory to grab a timeline trace. Ideally DevTools would integrate all the functionality we need, but it's not there yet.

Reply:
there isn't an option to turn it on off, so much as a page where you can opt into recording the samples. Maybe that is always on and the toggle is just whether or not they are shipped over to the tool.

Anyway, If we're always running the sampling profiler but just ignoring those samples, that seems wasteful.